src/users/users.service.ts

```diff
@@ -44,6 +65,24 @@ export class UsersService {
     return user;
   }
 
+  async hashPassword(password: string): Promise<string> {
+    // hash the password with SHA-256
+    const hash = createHash('sha256');
+    hash.update(password);
+    return hash.digest('hex');
+  }
```

Comment:
Please don't hash passwords just with sha-256.

Please let a proper library handle password hashing including salting, stretching and time resistant hashing functions such as argon2.

Reply:
It's no password in this way and salting is not needed for fully random byte strings because there can't be an effective rainbow table. The problem of using a hash function with salt is, that we don't know the user before we matched the key, so checking all keys would be needed.

Reply:
Then it shouldn't be called "hashPassword"? Also why not add a token identifier or use a JWT if you want to use the token itself as selector?

Maybe one needs to explain the concept of this to me, but for now, I'm not really seeing the goal of this token implementation.

Reply:
Yes renaming would make it clearer and it has to be commented. I think it's as secure as password auth, but maybe it's worth it to use JWTs.

Reply:
Initially the function was meant to be used for both passwords and tokens, but that changed in the mean time. A renaming will occur.

Regarding JWTs, maybe that would be an idea…